### ISSUE TITLE
[Suggestion] Escape usernames/input from users when mee6 posts them

### ISSUE THREAD

**Issue by smt923** [cookkkie/mee6#55]
Bit of a confusing title, don't know how exactly how to word it so here's an example:

Someone I know, a streamer with the username of DraQu has his twitch username exactly:

> DraQu_

The problem with this is _ is part of Discord's markdown, so when mee6 posts when he goes live, it turns into this: http://i.imgur.com/f2xyWso.png

You can see the first, non link underscore starts the italics, then ends it on the one from the twitch url, causing the twitch url to be incorrect (the rich preview style one works fine)

It's fairly minor, and I'm not sure where else it will be an issue (I suppose anywhere where a user could have a username or provide input that accepts characters from [Discord's markdown](https://support.discordapp.com/hc/en-us/articles/210298617-Markdown-Text-101-Chat-Formatting-Bold-Italic-Underline-) - but it would be a nice little thing to clean up, so that it won't be an issue for other people/confuse them if they're not sure exactly why their links don't work

I know a friend went into the support Discord and mentioned it, it was a while ago though and might have just been forgotten, just thought I'd leave an issue, again it's fairly minor so no big deal either way

**Comment by SilBoydens** [comment]
@smt923 we know this problem and there is not really a way that we can fix this, in the Mee6 support server we have a custom command that says:

(for streamers with a \_ in there name)
if there are 2 \_ in a message then all text in between wil be put cursif (discord layout system)
try changing {streamer} in \`{streamer}\`
then it should work
the message will then be :
hey all! `streamer_` just went online on <http://twitch.tv/streamer_>
(don't click that link)
because the streamername is then in a codeblock it won't recognise the \_ as markdown

you can try that to get it to work
~Sil from Mee6 support team
Mee6 support server: https://discord.gg/0tOgeGSG9kV0jasj

**Comment by Gstayton** [comment]
I don't know if the API the bot uses is significantly different or not, I only tested this personally on my own discord client as a quick test, but if you use \ to escape the first _, it works just fine. The one in the URL seems to not be escapable, though. 

name\_ in http://twitch.tv/kosan_ 
results in
http://i.imgur.com/YZ5r0UP.png

Is there any reason why this isn't implementable?

**Comment by smt923** [comment]
Yeah the first username is the only one that'd have to be escaped, I tried poking at the bot's code it's self, I obviously don't know the ins and outs so can't say if it's something down the line but it seems like it shouldn't be hard to find underscores when storing the username and just inserting an escape before the underscore in the username, but again, I don't know the ins and outs of the code it's self so maybe it would break something down the line that I haven't seen
